2024-july/demo-3/hospital-service/hospital_service.bal

```diff
@@ -0,0 +1,168 @@
+    # + return - A list of patients 
+    isolated resource function get patients() returns Patient[] {
+        lock {
+            return patientTable.toArray().clone();
+        }
+    }
+
+    # Get a patient by ID
+    #
+    # + return - returns can be any of following types 
+    # http:Ok (Patient details)
+    # http:NotFound (Patient not found)
+    isolated resource function get patients/[string patientId]() returns Patient|http:NotFound {
+        lock {
+            if !patientTable.hasKey(patientId) {
+                return {body: "Patient not found"};
+            }
+            return patientTable.get(patientId).clone();
+        }
+    }
+
+    # Create a new patient
+    #
+    # + patientEntry - PatientEntry object to be created 
+    # + return - Patient created successfully 
+    isolated resource function post patients(@http:Payload PatientEntry patientEntry) returns Patient {
+        Patient patient = {id: uuid:createRandomUuid(), ...patientEntry};
+        lock {
+            patientTable.put(patient.clone());
+        }
+        return patient;
+    }
+
+    # Update a patient by ID
+    #
+    # + patientEntry - Patient object to be updated 
+    # + return - returns can be any of following types 
+    # http:Ok (Patient updated successfully)
+    # http:NotFound (Patient not found)
+    isolated resource function put patients/[string patientId](@http:Payload PatientEntry patientEntry) returns Patient|http:NotFound {
+        lock {
+            if !patientTable.hasKey(patientId) {
+                return {body: "Patient not found"};
+            }
+        }
+        Patient patient = {id: patientId, ...patientEntry};
+        lock {
+            patientTable.put(patient.clone());
+        }
+        return patient;
+    }
+
+    # Delete a patient by ID
+    #
+    # + return - returns can be any of following types 
+    # http:NoContent (Patient deleted successfully)
+    # http:NotFound (Patient not found)
+    isolated resource function delete patients/[string patientId]() returns Patient|http:NotFound {
+        lock {
+            if !patientTable.hasKey(patientId) {
+                return {body: "Patient not found"};
+            }
+            return patientTable.remove(patientId).clone();
+        }
+    }
+
+    # Get a list of medical records
+    #
+    # + return - A list of medical records 
+    isolated resource function get records() returns MedicalRecord[] {
+        lock {
+            return recordTable.toArray().clone();
+        }
+    }
+
+    # Get a medical record by ID
+    #
+    # + return - returns can be any of following types 
+    # http:Ok (MedicalRecord details)
+    # http:NotFound (MedicalRecord not found)
+    isolated resource function get records/[string recordId]() returns MedicalRecord|http:NotFound {
+        lock {
+	        if !recordTable.hasKey(recordId) {
+	            return {body: "Record not found"};
+	        }
+	        return recordTable.get(recordId).clone();
```

Comment:
Shall we replace the tabs with spaces? Please check similar places. We can use a tool like [Refined GitHub](https://chromewebstore.google.com/detail/refined-github/hlepfoohegkhhmjieoechaddaejaokhf?hl=en) to see the white space characters clearly.

Reply:
Updated with c1ece123ac5f3b7661beb48fe4f5a5b93aa8d8c4